### PR TITLE
feat: deprecating middleware package

### PR DESCRIPTION
**Pull request by lukasborawski**
## Description
This PR provides info about `middleware` package deprecation.

The new middleware package was moved and active development is under [this](https://github.com/vuestorefront/middleware) repository.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

#### Code standards
- [x] My code follows the code style of this project
- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

File: packages/middleware/rollup.config.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import pkg from './package.json';
+import pkg from './_package.json';
```
Comment (Fifciu) [inline packages/middleware/rollup.config.js:1]: Why? Isn't there an option called `deprecated`? Also, we could set it to `private: true`, I mean inside the package.json, so it won't be possible to publish it anymore

File: packages/middleware/_package.json [hunk 2]
```diff
@@ -1,3 +1,9 @@
+/**
+  WARNING: Package deprecated (from development).
+  Do not provide any changes here.
+  Please check the https://github.com/vuestorefront/middleware repository,
+  where the current middleware package lives.
+*/
```
Comment (Fifciu) [inline packages/middleware/_package.json:6]: It's not a proper json file anymore. We shouldn't put comments here

Comment (Fifciu) [reply packages/middleware/_package.json:6]: No parser will handle it

**Review by Fifciu — CHANGES_REQUESTED** [review]

File: packages/middleware/_package.json (hunk 2 above)
Comment (lukasborawski) [reply packages/middleware/_package.json:6]: But why, you should not release it or parse it? That's the point.

**Comment by Mateuszpietrusinski** [comment]
Changes require more discussion, closing for now.

**Pull request closed without merging**